Packs/TAXIIServer/Integrations/TAXIIServer/TAXIIServer.py

```diff
@@ -933,27 +929,27 @@ def main():
     elif certificate and private_key:
         http_server = False
 
-    global SERVER
-    scheme = 'http'
-    host_name = server_link_parts.hostname
-    if not http_server:
-        scheme = 'https'
-
-    service_address = params.get('service_address')
-    SERVER = TAXIIServer(scheme, str(host_name), port, collections,
-                         certificate, private_key, http_server, credentials, service_address)
-
     demisto.debug(f'Command being called is {command}')
-    commands = {
-        'test-module': test_module
-    }
 
     try:
-        if command == 'long-running-execution':
+        port = get_port(params)
+        collections = get_collections(params)
+        server_links = demisto.demistoUrls()
+        server_link_parts: ParseResult = urlparse(server_links.get('server'))
+
+        global SERVER
+        scheme = 'http'
+        host_name = server_link_parts.hostname
+        if not http_server:
+            scheme = 'https'
+
+        service_address = params.get('service_address')
+        SERVER = TAXIIServer(scheme, str(host_name), port, collections,
+                             certificate, private_key, http_server, credentials, service_address)
+        if command == 'test-module':
+            return_results(test_module(SERVER))
+        elif command == 'long-running-execution':
             run_server(SERVER)
-        else:
```

Comment:
Why did you removed this?

Reply:
because it seems like there are no other commands then test_module and long-running-execution, so there is no "else" use case.  I put it back for future development, let me know otherwise.